qiskit/circuit/add_control.py

```diff
@@ -109,7 +108,7 @@ def control(operation: Union[Gate, ControlledGate],
                 use_basis_gates=True)
     elif operation.name == 'ry':
         qc.mcry(operation.definition[0][0].params[0], q_control, q_target[0],
-                q_ancillae, use_basis_gates=True)
+                None, mode='noancilla', use_basis_gates=True)
```

Comment:
@ewinston When I ran the tests, I got an error from this line (and L126, L134) due to an insufficient number of ancillas. The `mcry` (and `mct`) gate needs ancillas in the default mode if the number of controls is larger than 2. But `q_ancillae` is set to `None`, so these tests should've failed since they are also run with 3 and 4 qubits. 
I don't understand why they ran before, but I've updated them to use the `'noancilla'` mode. 
Do you know what's going on here?

Reply:
This passes tests because `mct` assumes `mode='noancilla'` if `q_ancilla` is `None`.
https://github.com/Qiskit/qiskit-terra/blob/6826b9d3c6a6ef13638ef4d6ec1c17d6493626cd/qiskit/extensions/standard/multi_control_toffoli_gate.py#L309. 

I did get failures when aqua was installed though.

Reply:
Ah I see, but with the updated Aqua version where these gates have been removed the tests should now pass. I reverted the changes here to use `q_ancillae` again.